engine/steps/lido/test/eth-to-steth.ts

```diff
@@ -0,0 +1,178 @@
+import { ADDRESS_ZERO, ASSET_TYPE_ERC20, ASSET_TYPE_NATIVE, AssetReference, createStandardProvider, EncodingContext, IERC20__factory, TEN_BIG } from '@freemarket/core'
+import { confirmTx, getTestFixture, getUsdt, MockWorkflowInstance, validateAction, WETH_ADDRESS } from '@freemarket/step-sdk/tslib/testing'
+import { WorkflowStruct } from '@freemarket/core/typechain-types/contracts/IWorkflowRunner'
+import { DepositEthForStEthAction } from '../typechain-types'
+
+
+const setup = getTestFixture(hre, async baseFixture => {
+  const {
+    signers: { otherUserSigner },
+    contracts: { frontDoor, userWorkflowRunner },
+    users: { otherUser },
+  } = baseFixture
+
+  // deploy the contract
+  await deployments.fixture('DepositEthForEZEthAction')
+
+  // get a reference to the deployed contract with otherUser as the signer
+  const depositEthForStEthAction = <DepositEthForStEthAction>await ethers.getContract('DepositEthForStEthAction', otherUserSigner)
+  const stEth = IERC20__factory.connect(MAINNET_STETH_ADDRESS, otherUserSigner)
+  const wstEth = IERC20__factory.connect(MAINNET_WSTETH_ADDRESS, otherUserSigner)
+
+  return {
+    contracts: { stEth, wstEth, depositEthForStEthAction, userWorkflowRunner, frontDoor },
+    hre,
+    otherUser,
+    otherUserSigner,
+  }
+})
+
+describe('Lido', async () => {
+  it('deploys', async () => {
+    const {
+      contracts: { configManager, depositEthForStEthAction },
+    } = await setup()
+    // simple sanity check to make sure that the action registered itself during deployment
+    await validateAction(configManager, STEP_TYPE_ID_LIDO_ETH_TO_STETH, depositEthForStEthAction.address)
+  })
+  it('Deposit ETH into stEth via runner', async () => {
+    const {
+      contracts: { stEth, wstEth, userWorkflowRunner },
+      otherUser,
+      otherUserSigner
+    } = await setup()
+    console.log(`otherUser ${otherUser}`)
+    const testAmount = ethers.utils.parseEther("0.001")
+    let stEthBefore = await stEth.balanceOf(otherUser)
+    const minStEth = testAmount.sub(100)
+    
+    const workflow: WorkflowStruct = {
+      workflowRunnerAddress: ADDRESS_ZERO,
+      steps: [
+        {
+          stepTypeId: STEP_TYPE_ID_LIDO_ETH_TO_STETH,
+          stepAddress: ADDRESS_ZERO,
+          inputAssets: [
+            {
+              sourceIsCaller: true,
+              amountIsPercent: false,
+              asset: {
+                assetType: ASSET_TYPE_NATIVE,
+                assetAddress: ADDRESS_ZERO,
+              },
+              amount: testAmount,
+            },
+          ],
+          argData: encodeDepositEthForStEthParams(testAmount.add(1)),
+          nextStepIndex: -1,
+        },
+      ],
+      beforeAll: [],
+      afterAll: [],
+    }
+
+    expect(userWorkflowRunner.executeWorkflow(workflow, {value : testAmount, gasLimit: 6000000})).to.be.reverted
```

Comment:
@0x-adam-smith these `expect(..).to.be.reverted` should be prefixed with `await` but when I add that I get unrecognized err:

HardhatChaiMatchersDecodingError: There was an error decoding '0000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000006408c379a000000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000019696e73756666696369656e74537445746852656365697665640000000000000000000000000000000000000000000000000000000000000000000000' as a string


TODO I need to figure out root cause. The await syntax looks correct and is used by other tests.